python/cudf/cudf/pandas/_wrappers/pandas.py

```diff
@@ -208,6 +214,12 @@ def _DataFrame__dir__(self):
     ]
 
 
+def custom_func(self, **kwargs):
```

Comment:
Is this to avoid checking `_ipython_canary_method_should_not_exist_` on both the cudf and pandas DataFrame? We should already be raising an AttributeError in the end?

Reply:
We are raising on `cudf` end but ipython does this weird thing of looping over all `repr` methods and checking if `_ipython_canary_method_should_not_exist_` exists every time, so it happens in a loop like:

`_ipython_canary_method_should_not_exist_`
`ipython_display`
`_ipython_canary_method_should_not_exist_`
`_repr_pdf_`
`_ipython_canary_method_should_not_exist_`
..
..
..
and so on. 
This looping is what was consuming all of the time.

Reply:
Ah ok thanks I didn't know ipython did that. Could we rename this function like `ignore_ipython_canary_check` to make this intent a little clearer?

Reply:
Sure

Reply:
Done.